WiseUnpacker/SabreTools.Serialization.Deserializers/WiseSection.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.IO;
+using System.Text;
+using SabreTools.IO.Extensions;
+using SabreTools.Models.WiseInstaller;
+using SabreTools.Matching;
+using static SabreTools.Models.WiseInstaller.Constants;
+
+namespace SabreTools.Serialization.Deserializers
+{
+    public class WiseSection : BaseBinaryDeserializer<WiseSectionHeader>
+    {
+        /// <inheritdoc/>
+        public override WiseSectionHeader? Deserialize(Stream? data)
+        {
+            // If the data is invalid
+            if (data == null || !data.CanRead)
+                return null;
+
+            // Cache the current offset
+            long initialOffset = data.Position;
+            try
+            {
+                var wiseSectionHeader = ParseWiseSectionHeader(data);
+
+                // Checks if version was able to be read
+                if (wiseSectionHeader.Version == null)
+                    return null;
+
+                // Main MSI file
+                if (wiseSectionHeader.MSIFileEntryLength == 0)
+                    return null;
+                else if (wiseSectionHeader.MSIFileEntryLength >= data.Length)
+                    return null;
+
+                // First executable file
+                if (wiseSectionHeader.FirstExecutableFileEntryLength == 0)
+                    return null;
+                else if (wiseSectionHeader.FirstExecutableFileEntryLength >= data.Length)
+                    return null;
+                
+                // Second executable file
+                if (wiseSectionHeader.SecondExecutableFileEntryLength == 0)
+                    return null;
+                else if (wiseSectionHeader.SecondExecutableFileEntryLength >= data.Length)
+                    return null;
+
+                return wiseSectionHeader;
+            }
+            catch
+            {
+                // Ignore the actual error
+                return null;
+            }
+        }
+        
+        
+        
+        /// <summary>
+        /// Parse a Stream into a WiseSectionHeader
+        /// </summary>
+        /// <param name="data">Stream to parse</param>
+        /// <returns>Filled WiseSectionHeader on success, null on error</returns>
+        private static WiseSectionHeader ParseWiseSectionHeader(Stream data)
+        {
+            var header = new WiseSectionHeader();
+            
+            int headerLength = -1; // indexed from 0
+            
+            // Find offset of "WIS", determine header length, read presumed version value
+            foreach (int offset in WisOffsets)
+            {
+                data.Seek(offset, 0);
+                byte[] checkBytes = data.ReadBytes(3);
+                if (checkBytes.EqualsExactly(WisString))
+                {
+                    headerLength = WiseSectionHeaderLengthDictionary[offset];
+                    data.Seek(offset - WiseSectionVersionOffsetDictionary[offset], 0);
```

Comment:
Nitpick: Please have the dictionary call on a separate line like the header length.